scripts/memory.py

```diff
@@ -1 +1,104 @@
-permanent_memory = []
+from config import Config, Singleton
+import pinecone
+import openai
+
+cfg = Config()
+
+
+def memory_factory(pinecone=True, openai=True, memory_name='auto-gpt', num_relevant=10):
+    """
+    Returns a memory object.
+    If pinecone is True and openai is True, returns a PineconeMemory object.
+    Otherwise, returns a SimpleMemory object.
+    OpenAI is to specify the OpenAI embeddings model
+    """
+    if not pinecone:
+        print('Using Simple Memory (no pinecone).')
+        return SimpleMemory()
+    if not openai:
+        print('Using Simple Memory (no openai). '
+              'If you want to use a different embeddings provider you must specify it in the config and impliment it!')
+        return SimpleMemory()
+    try:
+        pinecone_mem = PineconeMemory(memory_name, num_relevant)
+        # test the openai api since we need it to make embeddings.
+        get_ada_embedding("test")
+        return pinecone_mem
+    except Exception as e:
+        print(f"Error initializing memory {e}\nUsing Simple Memory.")
+        return SimpleMemory()
+
+
+def get_ada_embedding(text):
+    text = text.replace("\n", " ")
+    return openai.Embedding.create(input=[text], model="text-embedding-ada-002")["data"][0]["embedding"]
+
+
+class PineconeMemory(metaclass=Singleton):
+    def __init__(self, table_name, num_relevant):
+        pinecone_api_key = cfg.pinecone_api_key
+        pinecone_region = cfg.pinecone_region
+        pinecone.init(api_key=pinecone_api_key, environment=pinecone_region)
+        dimension = 1024
```

Comment:
Ada2 embeddings are 1536 - is there a reason you used 1024 here?
```suggestion
        dimension = 1536
```

Reply:
You are correct. Yeah, sorry, I ran into this the hard way. Should be fixed now.